src/docs/product/issues/issue-details/performance-issues/json-decoding-main-thread/index.mdx

```diff
@@ -0,0 +1,91 @@
+---
+title: "JSON Decoding on Main Thread"
+sidebar_order: 30
+redirect_from:
+  - /product/issues/performance-issues/json-decoding-main-thread/
+description: "Learn more about JSON Decoding on Main Thread issues and how to diagnose and fix them."
+---
+
+The main thread (or UI thread) in a mobile app is responsible for handling all user interaction, and maintaining a feeling of responsiveness requires the app to respond to gestures and taps in real time. When a long running operation blocks the main thread, the app becomes unresponsive, impacting the quality of the user experience.
```

Comment:
```suggestion
The main, (or UI) thread in a mobile app is responsible for handling all user interaction and needs to be able to respond to gestures and taps in real time. If a long-running operation blocks the main thread, the app becomes unresponsive, impacting the quality of the user experience.
```
Same comment as above. We can also write this if we want to be consistent with the File I/O on Main Thread language:
"The main UI thread in a mobile application handles user interface events such as button presses and page scrolls. To prevent App Hangs and Application Not Responding errors, the main UI thread should not be used for performing long-running operations. These kinds of operations block the whole UI until they finish running and get in the way of the user interacting with the app."